File: core/src/main/java/y2k/joyreactor/presenters/MessageThreadsPresenter.java
package y2k.joyreactor.presenters;

import y2k.joyreactor.Message;
import y2k.joyreactor.services.MessageService;

import java.util.List;

/**
 * Created by y2k on 01/10/15.
 */
public class MessageThreadsPresenter {

    public MessageThreadsPresenter(View view, MessageService service) {
        view.setIsBusy(true);
        service.get()
                .subscribe(threads -> {
                    view.setIsBusy(false);
                    view.reloadData(threads);
                }, Throwable::printStackTrace);
    }

    public void selectThread(int index) {
        // TODO:
//        Messenger.getInstance().send(new ThreadSelectedMessage(threads.get(index)));
    }

    public static class ThreadSelectedMessage {

        final Message thread;

        public ThreadSelectedMessage(Message thread) {
            this.thread = thread;
        }
    }

    public interface View {

        void setIsBusy(boolean isBusy);

        void reloadData(List<Message> threads);
    }
}

File: core/src/main/kotlin/y2k/joyreactor/presenters/MessageThreadsPresenter.kt
//package y2k.joyreactor.presenters
//
//import y2k.joyreactor.model.Message
//import y2k.joyreactor.common.subscribeOnMain
//import y2k.joyreactor.services.BroadcastService
//import y2k.joyreactor.services.UserMessagesService
//
///**
// * Created by y2k on 01/10/15.
// */
//class MessageThreadsPresenter(
//    private val view: MessageThreadsPresenter.View,
//    private val broadcastService: BroadcastService,
//    private val service: UserMessagesService) {
//
//    init {
//        view.setIsBusy(true)
//        service.getThreads()
//            .subscribeOnMain {
//                view.setIsBusy(false)
//                view.reloadData(it)
//            }
//    }
//
//    fun selectThread(thread: Message) {
//        broadcastService.broadcast(BroadcastService.ThreadSelectedMessage(thread))
//    }
//
//    interface View {
//
//        fun setIsBusy(isBusy: Boolean)
//
//        fun reloadData(threads: List<Message>)
//    }
//}